.github/workflows/ci.yml

```diff
@@ -114,47 +113,34 @@ jobs:
         uses: actions/cache@v3
         with:
           path: |
-            ~/.cabal/store
-
-          key: packages-cachebust-1-${{ hashFiles('cabal.project.freeze') }}
-          restore-keys: packages-cachebust-1-
-
-      - name: Install poetry
-        run: |
-          apt-get update
-          apt-get install python3-distutils -y
-          curl -sSL https://install.python-poetry.org | python3 -
-          echo "${HOME}/.local/bin" >> $GITHUB_PATH
-
-      - name: Install Python dependencies
-        run: |
-          cd elastic-buffer-sim && poetry update
+            ~/.cabal-nix/store
 
-      - name: Lint Python script
-        run: |
-          cd elastic-buffer-sim && poetry run pycodestyle genplots.py
+          key: packages-cachebust-2-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: packages-cachebust-2-
 
       # TODO: simulate + plot multiple topologies
       - name: Plot mesh grid
         run: |
-          cabal build elastic-buffer-sim:sim
-          cd elastic-buffer-sim && poetry run $(cabal list-bin elastic-buffer-sim:sim) +RTS -M8M -RTS plot complete3 150000 100
+          cabal run -- elastic-buffer-sim:sim +RTS -M8M -RTS plot complete3 150000 100
 
       - name: Upload plots
         uses: actions/upload-artifact@v2
         with:
           name: gen-plots-hs
           path: |
-            elastic-buffer-sim/_build/clockscomplete3.pdf
-            elastic-buffer-sim/_build/elasticbufferscomplete3.pdf
```

Comment:
Maybe I am missing someting here, but where do we enter the `elastic-buffer-sim` folder now. Since I cannot see any artifacts created by the CI Actions at the moment, this does not look like its working to me.

Reply:
Ok, the artifacts are there, I just was not on the summary view in github. Still wondering about the folder tough.

Reply:
Ah good point, I moved some thing around. I thought I reverted everything but I guess not!

Reply:
Are the artifacts than taken from the cache? Still wondering why this has not caused any error then?

Reply:
Sorry I mislead. Above these diff lines there's:

```diff
-          cabal build elastic-buffer-sim:sim
-          cd elastic-buffer-sim && poetry run $(cabal list-bin elastic-buffer-sim:sim) +RTS -M8M -RTS plot complete3 150000 100
+          cabal run -- elastic-buffer-sim:sim +RTS -M8M -RTS plot complete3 150000 100
```

An improvement IMO, but probably not something that should be part of this PR.

Reply:
Ok, the `_build` folder also gets created in the root folder. That was what I just did not recognice before. Thats fine then. These calls will be updated soon with the new simulation checks anyway.